packages/playwright-core/src/server/chromium/videoRecorder.ts

```diff
@@ -123,12 +124,17 @@ export class VideoRecorder {
     });
     this._process = launchedProcess;
     this._gracefullyClose = gracefullyClose;
+    this._launchTimestamp = monotonicTime();
   }
 
   writeFrame(frame: Buffer, timestamp: number) {
     assert(this._process);
     if (this._isStopped)
       return;
+    if (timestamp - this._lastFrameTimestamp < 1 / fps) {
```

Comment:
With the new backpressure mechanism added in https://github.com/microsoft/playwright/pull/9893 this should not be necessary. If the throttler doesn't give desired effect we should adjust its parameters instead. The difference with this approach is that Chromium won't send new screencast frames until it gets acks for already sent once whereas with the approach in this PR the frames are still generated just to be dropped on the floor on the playwright end.

Can you check if the problem is still reproducible on @next version (which includes the fix mentioned above)?